src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -1340,25 +1340,22 @@ public Tensor pin_memory()
             /// This attribute is null by default and becomes a Tensor the first time a call to backward() computes gradients for the tensor.
             /// The attribute will then contain the gradients computed and future calls to backward() will accumulate (add) gradients into it.
             /// </summary>
-            public Tensor? grad()
-            {
-                var res = NativeMethods.THSTensor_grad(Handle);
-                CheckForErrors();
-
-                if (res == IntPtr.Zero)
-                    return null;
+            public Tensor? grad {
+                get {
+                    var res = NativeMethods.THSTensor_grad(Handle);
 
-                return new Tensor(res);
-            }
+                    if (res == IntPtr.Zero) {
+                        CheckForErrors();
+                        return null;
+                    }
 
-            /// <summary>
-            /// This function will set the `tensor.grad()` attribute to a custom tensor. 
-            /// </summary>
-            /// <param name="grad">The new gradient tensor</param>
-            public void set_grad(Tensor grad)
-            {
-                NativeMethods.THSTensor_set_grad(Handle, grad?.DetachFromDisposeScope().Handle ?? IntPtr.Zero);
-                CheckForErrors();
+                    return new Tensor(res);
+                }
+                set {
+                    _ = value?.DetachFromDisposeScope();
```

Comment:
It seems it will automatically make the copy. I'm not sure:

```csharp
using System.Reflection;
using static TorchSharp.torch;

void DisposeButKeepHandle(Tensor tensor)
{
    var handle = tensor.Handle;
    tensor.Dispose();
    var f = tensor.GetType()
        .GetField("handle", BindingFlags.NonPublic | BindingFlags.Instance);
    f!.SetValue(tensor, handle);
}

var tensor = zeros([]);
var grad = zeros([]);
tensor.set_grad(grad);
DisposeButKeepHandle(grad);

Console.WriteLine(tensor.grad()! + 1);
// OK
Console.WriteLine(grad + 1);
// AccessViolationException: So the C++ tensor is true deleted.
// So tensor.grad (in the C++ level) is not the given grad?
```

Reply:
Yes, I have viewed the C++ code. It seems that it will use this operator to assign the value:

```cpp
  Tensor& operator=(const Tensor &x) & {
    return operator=(static_cast<const TensorBase&>(x));
  }
```

which uses:

```cpp
  Tensor& operator=(const TensorBase& x) & {
    impl_ = x.getIntrusivePtr();
    return *this;
  }
```

So `tensor.grad` will always be a new tensor. We can safely delete the original one.

Reply:
But how can we implement this behavior in Python:

```python
import torch

x = torch.zeros([])
y = torch.zeros([])

x.grad = y
print(id(y) == id(x.grad))
```

This is pretty strange. Perhaps PyTorch Tensor is using a different technique, instead of a simple pointer to LibTorch.

Reply:
Looks good!